src/Microsoft.Performance.SDK/Processing/ProcessingSource.cs

```diff
@@ -254,6 +279,43 @@ public virtual void DisposeProcessor(ICustomDataProcessor processor)
             IEnumerable<IDataSource> dataSources,
             IProcessorEnvironment processorEnvironment,
             ProcessorOptions options);
+        
+        /// <summary>
+        ///     When implemented in a derived class, creates a new
+        ///     instance implementing <see cref="ICustomDataProcessor"/>
+        ///     to process the specified <paramref name="dataSourceGroup"/>.
+        /// </summary>
+        /// <param name="dataSourceGroup">
+        ///     The <see cref="IDataSourceGroup"/> to be processed by the processor.
+        ///     This parameter is guaranteed to be non-null, and all
+        ///     elements in the collection are guaranteed to be non-null.
+        /// </param>
+        /// <param name="processorEnvironment">
+        ///     The environment for this specific processor instance.
+        /// </param>
+        /// <param name="options">
+        ///     The options to pass to the processor.
+        /// </param>
+        /// <returns>
+        ///     A new <see cref="ICustomDataProcessor"/>. It is an error
+        ///     to return <c>null</c> from this method.
+        /// </returns>
+        protected virtual ICustomDataProcessor CreateProcessorCore(
+            IDataSourceGroup dataSourceGroup,
+            IProcessorEnvironment processorEnvironment,
+            ProcessorOptions options)
+        {
+            if (this is IDataSourceGrouper)
+            {
+                throw new InvalidOperationException(
+                    $"Prior to V2, you must override the {nameof(CreateProcessorCore)} which accepts a {nameof(IDataSourceGroup)} if you implement {nameof(IDataSourceGrouper)}");
+            }
+            
+            this.Logger.Warn($"{this.GetType().Name} does not support processing user-specified processing groups - falling back to default processing.");
+	
+            // Call v1 methods for now
+            return this.CreateProcessor(dataSourceGroup.DataSources, processorEnvironment, options);
```

Comment:
Should we throw if dataSouceGroup.ProcessingMode is not DefaultProcessingMode?

Reply:
I'd prefer not to, since that would break backwards compatibility with plugins that don't implement this method if somehow that's the case. I think the only way for it to not be DefaultProcessingMode is if a driver/engine is doing something wrong. I'd rather the plugin silently use the V1 path in the case.

Perhaps a logger.Warn?

Reply:
Not in this case, since the check of `is IDataSourceGrouper` failed above. If we're hitting this line, it's an `IProcessingSource` that doesn't specify any processing modes.